apps/src/blockly/addons/cdoSerializationHelpers.js

```diff
@@ -32,11 +35,19 @@ function getXCoordinate(block, workspace) {
 
   // Multiplier accounts for the fact that blocks with SVG frames need twice as much padding
   // so their edges don't touch the edge of the workspace
-  let horizontalOffset = block.functionalSvg_ ? 2 * padding : padding;
+  let horizontalOffset = block.functionalSvg_
+    ? SVG_FRAME_SIDE_PADDING + padding
```

Comment:
This is an unrelated change (and I think desired) to the horizontal offset. Before, this was `2 * padding`, which is `2 * 16 = 32`. Now, I use padding + SVG frame padding, which is `16 + 15 = 31`. So functionally, this changes very little, but is more consistent with the logic for the vertical padding.

If the `viewWidth` is non-existent, this changes the `horizontalOffset` from `2 * 0 = 0` to `0 + 15 = 15`, but I'm not sure that's an edge case we should encounter in real life. Does that sound right?

Reply:
When could `viewWidth` be non-existent?

Reply:
It shouldn't be. It was a general comment on the first ternary here, which predates my work, IIRC. 
```
const padding = viewWidth ? WORKSPACE_PADDING : 0;
const width = viewWidth || contentWidth;
```

[viewWidth](https://developers.google.com/blockly/reference/js/blockly.utils_namespace.metrics_interface.viewwidth_propertysignature.md) from Blockly should always be a number. Maybe we need to fall back to content width if the workspace isn't visible? @mikeharv Do you have any additional context?

Reply:
I loaded `s/hello-world-food-2021/lessons/1/levels/6`. It looks fine, but this logic isn't hit because we don't need to adjust the block positions...

Reply:
I could change it to this if we're worried about the edge case where there's both no viewWidth (so padding is 0) and an SVG...
```
let horizontalOffset;
if (viewWidth && block.functionalSvg_) {
    horizontalOffset = SVG_FRAME_SIDE_PADDING + padding;
} else {
    horizontalOffset = padding;
}
```

Reply:
that seems like reasonable!

Reply:
@molly-moen @mikeharv I think these two commits should address the case above in a roundabout way: https://github.com/code-dot-org/code-dot-org/pull/55985/files/8998b3c0fb35874ab9da1b84de8ea5cfddb5608e..3dd1757a01779ce29988849222f80bebd235f76e